core/sdk-core/src/main/java/software/amazon/awssdk/core/internal/http/pipeline/stages/MakeAsyncHttpRequestStage.java

```diff
@@ -132,10 +156,12 @@ private ResponseHandler(Completable completable) {
 
         @Override
         public void headersReceived(SdkHttpResponse response) {
-            if (HttpStatusFamily.of(response.statusCode()) == HttpStatusFamily.SUCCESSFUL) {
+            if (response.isSuccessful()) {
                 isSuccess = true;
                 responseHandler.headersReceived(response);
+                SdkStandardLogger.REQUEST_LOGGER.debug(() -> "Received successful response: " + response.statusCode());
             } else {
+                SdkStandardLogger.REQUEST_LOGGER.debug(() -> "Received error response: " + response.statusCode());
                 errorResponseHandler.headersReceived(response);
```

Comment:
Can we consistently order these logs as happens-before or happens-after `headersReceived`?